File: Difficulty: Easy/Assign Mice Holes/assign-mice-holes.cpp
class Solution {
  public:
    int assignHole(vector<int>& mices, vector<int>& holes) {
        sort(mices.begin(), mices.end());
        sort(holes.begin(), holes.end());
        
        int ans = 0;
        for(int i=0; i<mices.size(); i++) {
            ans = max(ans, abs(mices[i] - holes[i]));
        }
        
        return ans;
    }
};

File: Difficulty: Easy/Assign Mice Holes/assign-mice-holes.java
class Solution {
    public int assignHole(int[] mices, int[] holes) {
        Arrays.sort(mices);
        Arrays.sort(holes);
        
        int ans = 0;
        for(int i=0; i<mices.length; i++) {
            ans = Math.max(ans, Math.abs(mices[i] - holes[i]));
        }
        
        return ans;
    }
};